library/core/src/array/iter.rs

```diff
@@ -84,6 +84,16 @@ impl<T, const N: usize> IntoIter<T, N> {
         IntoIterator::into_iter(array)
     }
 
+    /// Creates a new iterator from a partially initalized array.
+    ///
+    /// # Safety
+    ///
+    /// The caller must guarantee that all and only the `alive` elements of
+    /// `data` are initialized.
+    pub(crate) unsafe fn with_partial(data: [MaybeUninit<T>; N], alive: Range<usize>) -> Self {
```

Comment:
Looks like this isn't needed, and you can just call [`new_unchecked`](https://doc.rust-lang.org/nightly/std/array/struct.IntoIter.html#method.new_unchecked)?  (The method right below here.)

Reply:
Totally missed this thanks :joy:

Reply:
No worries -- it's pretty new.  I'm just extra familiar with it:

- #91341

(Having added it for exactly scenarios like this 🙃)